### ISSUE TITLE
test on AFLW2000 error is so large

### ISSUE THREAD

**Issue by flyduck** [natanielruiz/deep-head-pose#82]
Hi, firstly, very thanks for your greate work! I hava some question for this work:
(1)I coding and train a tensorflow model acording this work, when I test on the train dataset(300w-lp) its error is yaw=2.1 pitch=1.9 roll=1.7, but when I test the model on the test dataset(AFLW2000) its error is so large, yaw = 35.3 pitch=11.4 roll=12.5, I can't find where the problem~
(2)in AFLW2000 dataset, the face point have some cordinate is -1, how you crop the face roi use to test head pose? 
(3)in 300w-lp, the big pose face have error face point, how you process this dataset?

very thanks!

**Comment by CrossEntropy** [comment]
Hi, Have you solved this problem now？

**Comment by chuzcjoe** [comment]
> Hi, firstly, very thanks for your greate work! I hava some question for this work:
> (1)I coding and train a tensorflow model acording this work, when I test on the train dataset(300w-lp) its error is yaw=2.1 pitch=1.9 roll=1.7, but when I test the model on the test dataset(AFLW2000) its error is so large, yaw = 35.3 pitch=11.4 roll=12.5, I can't find where the problem~
> (2)in AFLW2000 dataset, the face point have some cordinate is -1, how you crop the face roi use to test head pose?
> (3)in 300w-lp, the big pose face have error face point, how you process this dataset?
> 
> very thanks!

Hi, have you solved the problem?

**Comment by flyduck** [comment]
> > Hi, firstly, very thanks for your greate work! I hava some question for this work:
> > (1)I coding and train a tensorflow model acording this work, when I test on the train dataset(300w-lp) its error is yaw=2.1 pitch=1.9 roll=1.7, but when I test the model on the test dataset(AFLW2000) its error is so large, yaw = 35.3 pitch=11.4 roll=12.5, I can't find where the problem~
> > (2)in AFLW2000 dataset, the face point have some cordinate is -1, how you crop the face roi use to test head pose?
> > (3)in 300w-lp, the big pose face have error face point, how you process this dataset?
> > very thanks!
> 
> Hi, have you solved the problem?



> Hi, Have you solved this problem now？
@CrossEntropy @chuzcjoe 
Hi, I have already solved these problem.
AFLW2000 dataset have some image the point is -1, so I dont use the point to get roi-image into heapose model, selective, I use face detect model, such as dlib, to detect face box and make the roi-image. This mothod used in the train data and test data also. That the train error and test error is more similar.

**Comment by CrossEntropy** [comment]
@flyduck Thanks for your help. I have solved this problem.

**Comment by chuzcjoe** [comment]
> > > Hi, firstly, very thanks for your greate work! I hava some question for this work:
> > > (1)I coding and train a tensorflow model acording this work, when I test on the train dataset(300w-lp) its error is yaw=2.1 pitch=1.9 roll=1.7, but when I test the model on the test dataset(AFLW2000) its error is so large, yaw = 35.3 pitch=11.4 roll=12.5, I can't find where the problem~
> > > (2)in AFLW2000 dataset, the face point have some cordinate is -1, how you crop the face roi use to test head pose?
> > > (3)in 300w-lp, the big pose face have error face point, how you process this dataset?
> > > very thanks!
> > 
> > 
> > Hi, have you solved the problem?
> 
> > Hi, Have you solved this problem now？
> > @CrossEntropy @chuzcjoe
> > Hi, I have already solved these problem.
> > AFLW2000 dataset have some image the point is -1, so I dont use the point to get roi-image into heapose model, selective, I use face detect model, such as dlib, to detect face box and make the roi-image. This mothod used in the train data and test data also. That the train error and test error is more similar.

Hi,
What is your final error on AFLW2000?

**Comment by flyduck** [comment]
@chuzcjoe 
about 5degrade error, but I use mobinetv2 as base model, and batch size is about 200.

**Comment by tengshaofeng** [comment]
@flyduck hi, I use the pretrained model hopenet_robust_alpha1.pkl  provide by author. And I use the dlib to detect face and test on AFLW2000. I get the following result:
##
Test error in degrees of the model on the 1969 test images. Yaw: 10.3898, Pitch: 7.5918, Roll: 6.1938
##
 What is the problem?

if not using dlib,  I get following result:
##
Test error in degrees of the model on the 1969 test images. Yaw: 9.1709, Pitch: 7.0361, Roll: 5.7762

##
what is the problem? Is it right?

**Comment by tengshaofeng** [comment]
> @flyduck Thanks for your help. I have solved this problem.
Can you share your training code? thanks in advance.
What is your hyper parameter like: alpha, learning rate ,batchsize, epoch and so on
